### PR TITLE
Feature/38 seb

### LINKED ISSUE #38: AH-6 & AH-9 Aiming Reticle
Add an aiming reticle to AH-6 and AH-9 aircraft.

Forum thread:
https://www.carpenoctem.co/forums/m/26081621/viewthread/33377293-inhouse-hud-mod-for-littlebirds
Rough demo:
https://discord.com/channels/154907081256730624/281495773894410240/789925912962531358

### PR DESCRIPTION
**Pull request by Seb105**
- Remove grease dot from AH-9
- Add collimator style aiming reticle to AH-6 and AH-9
- CBA settings to configure feature

### CODE REVIEW

**Review by milivojm — COMMENTED** [review]
I recommend moving rectacle.paa to cnto-assets to keep this clean and small. Images, sounds and other multimedia files are meant to be placed there and referenced here.

File: addons/littlebird_hud/fn_init.sqf [hunk 1]
```diff
@@ -0,0 +1,30 @@
+[
+	"a3aa_littlebird_hud_enable",
+	"CHECKBOX",
+	["Enable", "Enable small forward-facing collimator style reticle for AH-6 and AH-9 aircraft"],
+	["Arma Additions", "Littlebird HUD"],
+	true,  /* default */
+	false,  /* isGlobal */
+	nil,   /* script */
+	false   /* needRestart */
```
Comment (milivojm) [inline addons/littlebird_hud/fn_init.sqf:9]: Please double check if this argument is needed. Documentation (https://cbateam.github.io/CBA_A3/docs/files/settings/fnc_init-sqf.html) doesn't use it.

File: addons/littlebird_hud/fn_init.sqf [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+] call CBA_settings_fnc_init;
+if (!hasInterface) exitWith {};
```
Comment (milivojm) [inline addons/littlebird_hud/fn_init.sqf]: Hit enter before and after blocks for readability.

File: addons/littlebird_hud/fn_init.sqf [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+0 spawn {
+	waitUntil { !isNull player };
```
Comment (milivojm) [inline addons/littlebird_hud/fn_init.sqf:14]: Please add `sleep` for performance reasons.

File: addons/littlebird_hud/fn_init.sqf [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+	["vehicle", {
+		params ["_unit", "_newVehicle", "_oldVehicle"];
+		if (_newVehicle isKindOf "RHS_MELB_AH6M" OR _newVehicle isKindOf "Heli_Light_01_armed_base_F" && a3aa_littlebird_hud_enable) then {
```
Comment (milivojm) [inline addons/littlebird_hud/fn_init.sqf]: Not sure I understand your intent properly but `&&` will execute before `OR`. This means **RHS_MELB_AH6M** will always have the UI change applied, no matter the value of `a3aa_littlebird_hud_enable` setting. Was that on purpose or?

https://community.bistudio.com/wiki/Operators#Rules_of_Precedence

Comment (Seb105) [reply addons/littlebird_hud/fn_init.sqf]: Correct but I think that it is a leftover check, a3aa_littlebird_hud_enable is immediately checked by the called function anyway, so I will remove this unecessary check.

File: addons/littlebird_hud/fn_init.sqf (hunk 1 above)
Comment (Seb105) [reply addons/littlebird_hud/fn_init.sqf:9]: CBA Documentation is out of date. https://cbateam.github.io/CBA_A3/docs/files/settings/fnc_addSetting-sqf.html is the correct documentation, which actually just calls CBA_settings_fnc_init itself. Not sure why there are 2 functions that do the exact same thing (literally) with different documentation.

**Review by Seb105 — COMMENTED** [review]
> I recommend moving rectacle.paa to cnto-assets to keep this clean and small. Images, sounds and other multimedia files are meant to be placed there and referenced here.

Don't agree, files important to an addon should be found in close proximity to where they're actually used. This would create an unecessary dependency on cnto-assets

File: addons/littlebird_hud/fn_init.sqf (hunk 3 above)
Comment (Seb105) [reply addons/littlebird_hud/fn_init.sqf:14]: Nowhere else does cnto-additions do this. I've copied the code style from the rest of the addon, why change it here?

File: addons/littlebird_hud/fn_hud.sqf [hunk 5]
```diff
@@ -0,0 +1,29 @@
+if !(a3aa_littlebird_hud_enable) exitWith {};
+params ["_vehicle"];
+
+// AH-6 is zeroed to eye level of player, not vehicle centre. AH-9 is zeroed to centre of aircraft model. Therefore offset Z level of reticle if vehicle is AH-6
+private _offsetEye = [false,true] select (_vehicle isKindOf "RHS_MELB_AH6M");
```
Comment (milivojm) [inline addons/littlebird_hud/fn_hud.sqf]: `_vehicle isKindOf "RHS_MELB_AH6M"` returns a boolean itself.

**Review by milivojm — CHANGES_REQUESTED** [review]
Grease dot remains the problem, I've tried looking for it in the selections, object textures and model (no luck). It's not a custom display either nor a control attached to display 46. Think all that is left is trying to override properties of the class in the config file one by one until you luckily hit the one that adds that grease dot.

As far as paa location, since it is small, it can remain where it is now.

File: addons/littlebird_hud/config.cpp [hunk 6]
```diff
@@ -0,0 +1,27 @@
+class CfgPatches {
+    class littlebird_hud {
```
Comment (milivojm) [inline addons/littlebird_hud/config.cpp]: a3aa_ missing from class name.

**Comment by freghar** [comment]
(Not reviewing, but since I ran into a similar thing years ago, saving you some time here.)

You cannot easily remove the "grease dot". It's in `rhsusf_melb/Data/optics/melb_crosshair_dot_ca.paa` and was directly referenced when the `melb.p3d` model was built, which you would need to rebuild. There seems to also be a `.rvmat` for it, also directly referenced from `.p3d` (not via `hiddenSelections`, unfortunately).

There is
```
        class AnimationSources: AnimationSources {
            class Addcrosshair {
                source = "user";
                animPeriod = 1;
                initPhase = 1;
            };
```
but I doubt it's either related to the dot, or that the dot has more than one animation phase.

The code tries to set texture for something unrelated,
```
        hiddenSelections[] = {"camo1", "d_SN"};
        hiddenSelectionsTextures[] = {"rhsusf\addons\rhsusf_melb\data\melb_ext_co.paa", "rhsusf\addons\rhsusf_melb\data\decals\SN\blank_ca.paa"};
```
I'm not quite sure what "SN" means, there's quite a few `.paa`s with barely visible numbers (no clue what for), but the referenced `blank_ca.paa` is just a blank transparent image, not a grease dot.

It should be possible, I think, to override the RHS paa with a dedicated `pbo` which depends on `rhsusf_melb` (is loaded after it) and uses `rhsusf\addons\rhsusf_melb` as PBOPREFIX and has a blank/transparent `.paa` on the same path as the original one. However, I don't remember if this actually works or if you'd need to copy over all of the files. Alternatively, you could easily patch the existing RHS `pbo`, but it would need modification on every RHS update.

**Comment by milivojm** [comment]
I had some luck with grease dot. 

`lb animateSource ["Addcrosshair", 0, true];`

This seems to remove the dot by animating adding crosshair in phase 0 (no dot displayed). `lb` is my random variable for littlebird (in editor) so use something like `_vehicle` in your script.

If you want to show it just use 1 instead of 0.

**Review by milivojm — APPROVED** [review]
Nice! Thank you!

**Pull request merged**